packages/tizen_notification/tizen/src/tizen_notification_plugin.cc

```diff
@@ -40,478 +43,389 @@ class TizenNotificationPlugin : public flutter::Plugin {
-              FreeNotification(noti_handle);
-              LOG_ERROR("notification_set_sound failed : %s",
+      int32_t display_applist = 0;
+      if (GetValueFromEncodableMap(arguments, "displayApplist",
+                                   display_applist)) {
+        ret = notification_set_display_applist(handle, display_applist);
+        if (ret != NOTIFICATION_ERROR_NONE) {
+          FreeNotification(handle);
+          result->Error("notification_set_display_applist failed",
                         get_error_message(ret));
-              result->Error("notification_set_sound failed",
-                            std::string(get_error_message(ret)));
-              return;
-            }
-          }
+          return;
         }
+      }
 
-        flutter::EncodableValue vibration;
-        if (GetEncodableValueFromArgs(arguments, "vibration", vibration)) {
-          std::string type;
-          if (GetValueFromArgs(&vibration, "type", type)) {
-            std::string path;
-            char *resource_path = app_get_shared_resource_path();
-            if (GetValueFromArgs(&vibration, "path", path)) {
-              path = std::string(resource_path) + path;
-            }
-            ret = notification_set_vibration(
-                noti_handle, StringToVibrationType(type), path.c_str());
-            free(resource_path);
-            if (ret != NOTIFICATION_ERROR_NONE) {
-              FreeNotification(noti_handle);
-              LOG_ERROR("notification_set_vibration failed : %s",
+      int32_t properties = 0;
+      if (GetValueFromEncodableMap(arguments, "properties", properties)) {
+        ret = notification_set_property(handle, properties);
+        if (ret != NOTIFICATION_ERROR_NONE) {
+          FreeNotification(handle);
+          result->Error("notification_set_property failed",
                         get_error_message(ret));
-              result->Error("notification_set_vibration failed",
-                            std::string(get_error_message(ret)));
-              return;
-            }
-          }
+          return;
         }
+      }
 
-        if (GetEncodableValueFromArgs(arguments, "appControl",
-                                      app_control_data)) {
-          std::string app_id;
-          if (GetValueFromArgs(&app_control_data, "appId", app_id)) {
-            std::string operation;
-            std::string uri;
-            std::string category;
-            std::string mime;
-            flutter::EncodableMap extras;
-            flutter::EncodableValue extra_value;
-            std::string key;
-            ret = app_control_create(&app_control);
-            if (ret != APP_CONTROL_ERROR_NONE) {
-              FreeNotification(noti_handle);
-              LOG_ERROR("app_control_create failed : %s",
+      flutter::EncodableMap sound;
+      if (GetValueFromEncodableMap(arguments, "sound", sound)) {
+        std::string type;
+        GetValueFromEncodableMap(&sound, "type", type);
```

Comment:
This case and the "vibration" case do not handle false return.
Is it an intentional?
(use `NOTIFICATION_{SOUND/VIBRATION}}_TYPE_NONE` for invalid argument(?))
If not, how about adding a casefor Invalid argument?

Reply:
Not setting the `sound` argument or setting it to null will result in `notification_set_sound` not being called and the notification module will fall back to its default value (`NOTIFICATION_SOUND_TYPE_NONE` according to the documentation). Currently if the `NotificationSound` property is not set in the Dart side, a null value will be passed (it's a valid input) and the default sound will be used. So yes, this is an intended behavior.